metals/src/main/scala/scala/meta/internal/metals/codeactions/AddingBracesCodeAction.scala

```diff
@@ -0,0 +1,641 @@
+            val distanceToThen = (cursorLine - tryLine).abs
+            val distanceToElseP = (cursorLine - Try(
+              termIf.elsep.pos.toLSP.getStart.getLine
+            ).getOrElse(Int.MaxValue)).abs
+            Try(
+              Seq(
+                (
+                  createCodeActionForPotentialBlockHolder(
+                    termIf,
+                    path,
+                    document,
+                    termIf.thenp,
+                    "then expression",
+                    "then"
+                  ),
+                  distanceToThen
+                ),
+                (
+                  createCodeActionForPotentialBlockHolder(
+                    termIf,
+                    path,
+                    document,
+                    termIf.elsep,
+                    "else expression"
+                  ),
+                  distanceToElseP
+                )
+              ).minBy(_._2)
+            ).map(_._1).toOption.flatten.toList
+
+          case termFor: Term.For =>
+            createCodeActionForPotentialBlockHolder(
+              termFor,
+              path,
+              document,
+              termFor.body,
+              "for expression"
+            ).toSeq
+          case termForYield: Term.ForYield =>
+            createCodeActionForPotentialBlockHolder(
+              termForYield,
+              path,
+              document,
+              termForYield.body,
+              "yield expression"
+            ).toSeq
+          case termWhile: Term.While =>
+            createCodeActionForPotentialBlockHolder(
+              termWhile,
+              path,
+              document,
+              termWhile.body,
+              "while expression",
+              "do"
+            ).toSeq
+          case _: Defn.GivenAlias => None.toSeq
+          case template: Template =>
+            val title = template.parent
+              .collectFirst {
+                case _: Defn.Enum => "enum definition"
+                case _: Defn.Trait => "trait definition"
+                case _: Defn.Object => "object definition"
+                case _: Defn.Class => "class definition"
+              }
+              .getOrElse("template")
+            template.parent.flatMap {
+              createCodeActionForTemplateHolder(
+                _,
+                path,
+                document,
+                template,
+                title
+              )
+            }.toSeq
+          case termBlock: Term.Block if !termBlock.parent.exists(_ match {
+                case _: Term.Apply => true
+                case _ => false
+              }) =>
+            termBlock.parent.flatMap { blockHolder =>
+              createCodeActionForPotentialBlockHolder(
+                blockHolder,
+                path,
+                document,
+                termBlock,
+                "block"
+              )
+            }
+          case termMatch: Term.Match =>
```

Comment:
Could we drop Match and Catch for now and focus on templates and blocks?

Reply:
Yes, sure.